protos/core.proto

```diff
@@ -26,22 +26,62 @@ message ProjectConfig {
   string gcloud_project_id = 6;
 }
 
-message Environment {
-  string name = 1;
+message Environments {
+  message Environment {
```

Comment:
I'm not a fan of this, it's going to make all our imports/Types horribly long for no obvious benefit?
Would it not be better to put this in a new namespace?

Reply:
We can put it in a different namespace, that's fine with me, but I do think it was a mistake not to nest `Environment` inside `Environments` in the first place. 

For me at least, nesting messages inside the scope where they're used feels much cleaner, and that makes it more readable/understandable.

I actually wish that more of these protobufs were nested, I often find it hard to navigate our protobuf files and figure out what messages are used where.

Reply:
Also I don't really think we should structure protobufs to suit any specific target language. If our TS types become verbose, there are workarounds for that, e.g. type aliases.